### DISCUSSION TITLE
Why isn't this a part of vs code?

### DISCUSSION THREAD

**Discussion by gabimor** [yoavbls/pretty-ts-errors#34] [General]
Really weird that this needs an extension to be honest. TS is a de facto standard nowadays, why did Yoav need to go through
all this trouble (and be so creative to bypass VS code limitations) and how long before VS Code team picks up on the scent and give us all first class support
for nicer TS errors? (And well deserved credit to pretty-ts-errors)

**Comment by yoavbls** [comment]
Thank you @gabimor!
Why they didn't implement it themself is a good question...
VSCode is getting the errors as a raw string from TypeScript LSP. If VSCode would support rich text/markdown for diagnostics (like this [issue](https://github.com/microsoft/vscode/issues/54272) suggests) maybe TypeScript LSP could return a formatted text for IDE's that are willing too

**Reply by rchl** [reply]
As you said, it's Typescript that returns the error message so I would say that the first step towards improving the error messages formatting would be to start at the root which is [TypeScript](https://github.com/microsoft/TypeScript) itself. If they would agree to improve the formatting in the way your package does then it would already be at least 50% success.

This would still not allow messages to have markdown formatting since the LSP spec doesn't support that but if the whitespace formatting would be handled by Typescript then this package would have to do a lot less itself.
